src/main/java/org/researchspace/services/storage/s3/S3Storage.java

```diff
@@ -0,0 +1,190 @@
+            .withCredentials(new AWSStaticCredentialsProvider(awsCreds))
+            .build();
+        
+    }
+
+    @Override
+    public boolean isMutable() {
+        return config.isMutable();
+    }
+
+    @Override
+    public Optional<ObjectRecord> getObject(StoragePath path, String revision) throws StorageException {
+
+        try {
+            Optional<String> key = this.paths.mapForward(path).map(StoragePath::toString);
+
+            if (!key.isPresent())
+                return Optional.empty(); 
+
+            com.amazonaws.services.s3.model.ObjectMetadata s3Metadata = s3.getObjectMetadata(config.getBucket(), key.get());
+            
+            // If metadata exist
+            String user = s3Metadata.getUserMetaDataOf("Author");
+            S3StorageLocation location = new S3StorageLocation(config.getBucket(), key.get());
+            ObjectMetadata metadata = new ObjectMetadata(Objects.isNull(user) ? "" : user, s3Metadata.getLastModified().toInstant());
+
+            ObjectRecord record = new ObjectRecord(location, path, revision, metadata);
+
+            return Optional.of(record);        
+        } 
+        catch (AmazonS3Exception e) {
+            // If the key is missing, return an empty result
+            if (StringUtils.equals(e.getErrorCode(), "NoSuchKey") || StringUtils.equals(e.getErrorCode(), "404 Not Found"))
+                return Optional.empty();
+            throw new StorageException(e.getMessage());
+        } 
+        catch (Exception e) {
+            throw new StorageException(e.getMessage());
+        }
+
+    }
+    
+
+    @Override
+    public List<ObjectRecord> getRevisions(StoragePath path) throws StorageException {
+        // TODO Auto-generated method stub
+        return null;
+    }
+
+    @Override
+    public List<ObjectRecord> getAllObjects(StoragePath prefix) throws StorageException {
+
+        Optional<StoragePath> mappedPrefix = paths.mapForward(prefix);
+        
+        String pref = mappedPrefix.get().toString();
+        
+        try {
+
+            ListObjectsV2Result result = s3.listObjectsV2(config.getBucket(), pref);
+            List<ObjectRecord> records = new ArrayList<>();
+
+            List<S3ObjectSummary> objects = result.getObjectSummaries();
+            for (S3ObjectSummary os : objects) {
+
+                Optional<StoragePath> path = StoragePath.tryParse(os.getKey()).flatMap(paths::mapBack);
+                Optional<ObjectRecord> record = getObject(path.get(), null);
+
+                if (record.isPresent())
+                    records.add(record.get());
+            }
+
+            return records;
+    
+        } catch (Exception e) {
+
+            throw new StorageException(e.getMessage());
+        }
+    }
+
+    @Override
+    public ObjectRecord appendObject(StoragePath path, ObjectMetadata metadata, InputStream content, long contentLength)
+            throws StorageException {
+        // TODO Auto-generated method stub
```

Comment:
can we have implementation for append and delete, I believe it should be quite straightforward to do:

https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/AmazonS3.html#putObject-java.lang.String-java.lang.String-java.io.InputStream-com.amazonaws.services.s3.model.ObjectMetadata-

https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/AmazonS3.html#deleteObject-java.lang.String-java.lang.String-